tools/clang/test/CodeGenSPIRV/texture.sample-cmp.hlsl

```diff
@@ -0,0 +1,37 @@
+// Run: %dxc -T ps_6_0 -E main
+
+SamplerComparisonState gSampler : register(s1);
+
+Texture1D   <float4> t1 : register(t1);
+Texture2D   <float2> t2 : register(t2);
+TextureCube <float>  t4 : register(t4);
+// No .SampleCmp() for Texture3D.
+
+// CHECK: [[v2fc:%\d+]] = OpConstantComposite %v2float %float_0_1 %float_0_2
+// CHECK: [[v3fc:%\d+]] = OpConstantComposite %v3float %float_0_1 %float_0_2 %float_0_3
+
+float4 main(int2 offset: A, float comparator: B) : SV_Target {
+// CHECK:              [[t1:%\d+]] = OpLoad %type_1d_image %t1
+// CHECK-NEXT:   [[gSampler:%\d+]] = OpLoad %type_sampler %gSampler
+// CHECK-NEXT: [[comparator:%\d+]] = OpLoad %float %comparator
+// CHECK-NEXT: [[sampledImg:%\d+]] = OpSampledImage %type_sampled_image [[t1]] [[gSampler]]
+// CHECK-NEXT:            {{%\d+}} = OpImageSampleDrefImplicitLod %float [[sampledImg]] %float_0_1 [[comparator]] ConstOffset %int_5
```

Comment:
OpImageSampleDrefImplicitLod :
> Result Type must be a scalar of integer type or floating-point type. It must be the same as Sampled Type of the underlying OpTypeImage.

Maybe change one of these to return `int` type?

Reply:
Same here. Cannot sample from non-float textures.

Reply:
Aah! I see :)